detect_secrets/plugins/oauth.py

```diff
@@ -0,0 +1,12 @@
+import re
+
+from .base import RegexBasedDetector
+
+
+class BasicOAuthDetector(RegexBasedDetector):
```

Comment:
```suggestion
class SquareOAuthDetector(RegexBasedDetector):
```

Since this is specifically relating to square secrets, let's just call it as such. Please also rename the file accordingly.

Reply:
Done! Thank you very much @domanchi